package/utils/uencrypt/Makefile

```diff
@@ -0,0 +1,57 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2022 Eneas Ulir de Queiroz
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=uencrypt
+PKG_RELEASE:=1
+
+PKG_FLAGS:=nonshared
+PKG_LICENSE:=GPL-2.0-or-later
+PKG_MAINTAINER:=Eneas U de Queiroz <cotequeiroz@gmail.com>
+PKG_CONFIG_DEPENDS:=\
+	CONFIG_UENCRYPT_OPENSSL \
+	CONFIG_UENCRYPT_WOLFSSL
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+CMAKE_INSTALL:=1
+CMAKE_OPTIONS+=$(if $(CONFIG_UENCRYPT_WOLFSSL),-DUSE_WOLFSSL=1)
+
+define Package/uencrypt
+  SECTION:=utils
+  CATEGORY:=Base system
+  TITLE:=Decryption utility for Arcadyan WG4xx223
+  DEPENDS:=@TARGET_ramips_mt7621 +UENCRYPT_WOLFSSL:libwolfssl +UENCRYPT_OPENSSL:libopenssl
```

Comment:
I wonder how this is handled when someone select openssl instead of wolfssl?
Should we have a a variant system and create 2 different package for openssl and wolfssl ?

Reply:
I though that the wolfSSL version would be enough, but since I used OpenSSL API, I thought why not allow both?

I'm assuming most people installing openssl would have the openssl util installed as well, and the script can easily handle both.  In that case, it would just waste 2.4kb of flash.  I'll suggest the use of openssl util in #10233.

The way I have made it, you have to switch providers at `menuconfig`.  Packaging separate variants would allow installing it through the image builder.

I can make it work either way.  It's your call.

Reply:
Yhea good point... if someone have big openssl he should have space for openssl-utils and use that instead...
We just have to take extra care in any script using this tool to also support openssl-util as an alternative...